e2e/config/vhd.go

```diff
@@ -114,29 +185,58 @@ func findLatestImageWithTag(imageID, tagName, tagValue string) (VHDResourceID, e
 	if latestVersion == nil {
 		return "", ErrNotFound
 	}
+
+	if err := ensureReplication(ctx, client, definition, latestVersion); err != nil {
+		return "", fmt.Errorf("ensuring image replication: %w", err)
+	}
+
 	return VHDResourceID(*latestVersion.ID), nil
 }
 
-type imageID struct {
-	subscriptionID string
-	resourceGroup  string
-	galleryName    string
-	imageName      string
+func ensureReplication(ctx context.Context, client *armcompute.GalleryImageVersionsClient, definition sigImageDefinition, version *armcompute.GalleryImageVersion) error {
+	if replicatedToCurrentRegion(version) {
+		return nil
+	}
+	return replicateToCurrentRegion(ctx, client, definition, version)
+}
+
+func replicatedToCurrentRegion(version *armcompute.GalleryImageVersion) bool {
+	for _, targetRegion := range version.Properties.PublishingProfile.TargetRegions {
+		if strings.EqualFold(strings.ReplaceAll(*targetRegion.Name, " ", ""), Location) {
+			return true
+		}
+	}
+	return false
 }
 
-func parseImageID(resourceID string) (imageID, error) {
-	pattern := `(?i)^/subscriptions/([^/]+)/resourceGroups/([^/]+)/providers/Microsoft\.Compute/galleries/([^/]+)/images/([^/]+)$`
-	re := regexp.MustCompile(pattern)
-	matches := re.FindStringSubmatch(resourceID)
+func replicateToCurrentRegion(ctx context.Context, client *armcompute.GalleryImageVersionsClient, definition sigImageDefinition, version *armcompute.GalleryImageVersion) error {
+	log.Printf("will replicate image version %s to region %s...", *version.ID, Location)
 
-	if matches == nil || len(matches) != 5 {
-		return imageID{}, fmt.Errorf("failed to parse image ID %q", resourceID)
+	version.Properties.PublishingProfile.TargetRegions = append(version.Properties.PublishingProfile.TargetRegions, &armcompute.TargetRegion{
+		Name:                 &Location,
+		RegionalReplicaCount: to.Ptr[int32](1),
+		StorageAccountType:   to.Ptr(armcompute.StorageAccountTypeStandardLRS),
+	})
+
+	resp, err := client.BeginCreateOrUpdate(ctx, definition.resourceGroup, definition.gallery, definition.definition, *version.Name, *version, nil)
+	if err != nil {
+		return fmt.Errorf("begin updating image version target regions: %w", err)
+	}
+	if _, err := resp.PollUntilDone(ctx, nil); err != nil {
+		return fmt.Errorf("updating image version target regions: %w", err)
 	}
 
-	return imageID{
-		subscriptionID: matches[1],
-		resourceGroup:  matches[2],
-		galleryName:    matches[3],
-		imageName:      matches[4],
-	}, nil
+	return nil
+}
+
+func newGalleryImageVersionsClient(subscriptionID string) (*armcompute.GalleryImageVersionsClient, error) {
```

Comment:
currently all azure clients are in `config.Azure`.

Reply:
ahh good catch, will rework

Reply:
fixed